docs/research/user-experience/xmr-btc-user-interface-definition.md

```diff
@@ -0,0 +1,145 @@
+# XMR<>BTC Atomic Swap - User Interface Prototypes
+
+This document specifies assumptions concerning possible user interfaces on top of the [Monero-Bitcoin swap as described by Lucas on the COMIT-blog](https://comit.network/blog/2020/10/06/monero-bitcoin/).
+
+We first specify assumptions and limitations imposed by the current setup and software needed for the swap, then outline possible solutions. 
+This document will be used as a basis for defining and evaluating low-fidelity UI prototypes in the prototyping tool Figma.
+
+## Assumptions
+
+This section sums up assumptions around the current setup, protocol and existing software. 
+
+### Technical Limitations
+
+Swap-solutions that run completely in the browser (webpage) can be problematic because blockchain nodes, notably [bitcoind](https://github.com/bitcoin/bitcoin/issues/11833), do not specifying [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) in their response.
```

Comment:
My opinion on this matter is that it is a feature, not a technical limitation. I would not want random downloaded javascript to control my bitcoind, even if we provide a way to verify the JS, users usually don't and then someone could run a single market maker website with malicious JS that steal coins.

The client that access bitcoind/monerod should be explicitly downloaded by the user. Whether it's a desktop app (self built or signed) or an browser extension (signed).

I'd be more keen to work with this from a security point of view than "boo bitcoind restricts us" :)

Reply:
I re-worded it.